src/book/book.service.ts

```diff
@@ -68,6 +69,11 @@ export class BookService {
       }
     }
 
+    // Given that we get a string from the query, we need to convert it to a boolean
+    if (isDeleted === 'true' || isDeleted === 'false') {
```

Comment:
I suggest that the related logic to the filterconditions move it to a separate function called setFilters